src/crypto.ts

```diff
@@ -53,3 +53,33 @@ function encodeString(str: string): Uint8Array {
   encoded.set(stringToUint8Array(str), 8);
   return encoded;
 }
+
+export function deriveChildSeed(masterSeed: string, seed: string): string {
+  return HashAll(masterSeed, seed).toString();
+}
+
+export function generateKeyPairAndSeed(length?: 64): { publicKey: PublicKey; privateKey: SecretKey; seed: string } {
+  const seed = makeSeed(length);
+  const { publicKey, privateKey } = keyPairFromSeed(seed);
+  return { publicKey, privateKey, seed };
+}
+
+/**
+ * Generates a public and private key from a provided, secure seed.
+ * @param seed - A secure seed.
+ */
+export function keyPairFromSeed(seed: string): { publicKey: PublicKey; privateKey: SecretKey } {
+  // Get a 32-byte seed.
+  seed = pkcs5.pbkdf2(seed, "", 1000, 32, md.sha256.create());
+  return pki.ed25519.generateKeyPair({ seed });
+}
+
+function makeSeed(length: number) {
+  let result = "";
+  const characters = "abcdefghijklmnopqrstuvwxyz";
+  const charactersLength = characters.length;
+  for (let i = 0; i < length; i++) {
+    result += characters.charAt(Math.floor(Math.random() * charactersLength));
+  }
+  return result;
+}
```

Comment:
That function seems weak for a seed generator 
- there is a chance that `Math.floor(Math.random() * charactersLength)` returns an index out of bounds if `Math.random()` returns `1`
- `Math.random()` is not a cryptographically safe random function - there is a number of articles on the internet about it

We should use a safer algorithm for a random seed generation.

Reply:
For SkyID I used this, maybe it helps: https://github.com/escada-finance/sia-js/blob/afbdb7eb8edbb050c2ac7272d243b02fd5604a25/lib/KeyPair.js#L23
It is because I need to generate a mnemonic.

Reply:
Thanks @DaWe35, I just saw your comment. I implemented it only for browser and looks like it fails when running the tests in Node. Will nick your function 😉